core/dbt/adapters/base/impl.py

```diff
@@ -475,14 +472,23 @@ def _make_match(self, relations_list, schema, identifier):
         return matches
 
     @available
-    def get_relation(self, schema, identifier, model_name=None):
-        relations_list = self.list_relations(schema, model_name)
+    def get_relation(self, database, schema, identifier, model_name=None):
+        assert schema is not None
+        assert database is not None
```

Comment:
what's the end user behavior here if schema or database are `None`?

we should take care introducing runtime assertions. I could get comfortable with the idea, but it feels like you're adopting a new practice here

Reply:
I'l just remove these, they were mostly for initial testing work and tracking down an "impossible" bug. With these changes, database/schema really shouldn't be None here ever, but dbt will do the right thing anyway.